grails-app/controllers/org/pih/warehouse/inventory/StockTransferController.groovy

```diff
@@ -97,4 +98,34 @@ class StockTransferController {
         }
         redirect(action: "list")
     }
+
+    def deleteInboundOrOutboundReturn = {
+        Location currentLocation = Location.get(session.warehouse.id)
+        Order orderInstance = Order.get(params.orderId)
+
+        // Inbound return case
+        if (orderInstance?.getStockMovementDirection(currentLocation) == StockMovementDirection.INBOUND) {
+            redirect(controller: "stockMovement", action: "list", params: ['direction': StockMovementDirection.INBOUND])
+            try {
+                stockTransferService.deleteStockTransfer(params.orderId)
+                stockMovementService.deleteStockMovement(params.id)
```

Comment:
You should delete the shipment in the `deleteStockTransfer` (if order has one) instead of triggering `stockMovementService.deleteStockMovement(params.id)`

You have to add this in the `deleteStockTransfer`:
```
order.shipments?.each { Shipment it -> 
   shipmentService.deleteShipment(shipment)
}
```